joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/GoSrc2Cpg.scala

```diff
@@ -20,6 +24,13 @@ class GoSrc2Cpg extends X2CpgFrontend[Config] {
       File.usingTemporaryDirectory("gosrc2cpgOut") { tmpDir =>
         new MetaDataPass(cpg, Languages.GOLANG, config.inputPath).createAndApply()
         val astGenResult = new AstGenRunner(config).execute(tmpDir)
+        GoMod.config = Some(config)
+        (if (!astGenResult.parsedModFile.isEmpty)
```

Comment:
Could this code not simply be: 
```scala
astGenResult.parsedModFile.foreach(modFile => 
    GoAstJsonParser.readModFile(Paths.get(modFile)).foreach(x => GoMod.meta = x)
)

Reply:
There can be only one go.mod file present. I guess I could simplify it as following.

```
        GoMod.meta =
          if (!astGenResult.parsedModFile.isEmpty)
            GoAstJsonParser.readModFile(Paths.get(astGenResult.parsedModFile.get))
          else None
```

Reply:
The `Option` class implements `foreach` as a means to operate on the value of it exists, so no need to do a `isDefined` check or `get` call.

Reply:
done